OpenRA.Mods.Common/Widgets/TextFieldWidget.cs

```diff
@@ -115,82 +120,180 @@ public override bool HandleKeyPress(KeyInput e)
 			if (!HasKeyboardFocus)
 				return false;
 
-			if ((e.Key == Keycode.RETURN || e.Key == Keycode.KP_ENTER) && OnEnterKey())
-				return true;
+			Func<int> getPrevWhitespaceIndex = () =>
+				Text.Substring(0, CursorPosition).TrimEnd().LastIndexOf(' ') + 1;
+
+			Func<int> getNextWhitespaceIndex = () => {
+				var substr = Text.Substring(CursorPosition);
+				var substrTrimmed = substr.TrimStart();
+				var trimmedSpaces = substr.Length - substrTrimmed.Length;
+				var nextWhitespace = substrTrimmed.IndexOf(' ');
+				if (nextWhitespace == -1)
+					return Text.Length;
+				else
+					return CursorPosition + trimmedSpaces + nextWhitespace;
+			};
+
+			var isOSX = Platform.CurrentPlatform == PlatformType.OSX;
+
+			switch (e.Key) {
+				case Keycode.RETURN:
+				case Keycode.KP_ENTER:
+					if (OnEnterKey())
+						return true;
+					break;
 
-			if (e.Key == Keycode.TAB && OnTabKey())
-				return true;
+				case Keycode.TAB:
+					if (OnTabKey())
+						return true;
+					break;
 
-			if (e.Key == Keycode.ESCAPE && OnEscKey())
-				return true;
+				case Keycode.ESCAPE:
+					if (OnEscKey())
+						return true;
+					break;
 
-			if (e.Key == Keycode.LALT && OnAltKey())
-				return true;
+				case Keycode.LALT:
+					if (OnAltKey())
+						return true;
+					break;
 
-			if (e.Key == Keycode.LEFT)
-			{
-				if (CursorPosition > 0)
-					CursorPosition--;
+				case Keycode.LEFT:
+					ResetBlinkCycle();
+					if (CursorPosition > 0)
+						if ((!isOSX && e.Modifiers.HasModifier(Modifiers.Ctrl)) ||
+						(isOSX && e.Modifiers.HasModifier(Modifiers.Alt)))
+							CursorPosition = getPrevWhitespaceIndex();
+						else
+							CursorPosition--;
+					break;
 
-				return true;
-			}
+				case Keycode.RIGHT:
+					ResetBlinkCycle();
+					if (CursorPosition <= Text.Length - 1)
```

Comment:
Please add braces around the body of this outer if.  While the if/else is a single statement (so its not technically wrong), the potential for confusion about what the else belongs to means it always pays to be explicit in these cases.